server: TestServerWithTimeseriesImport failed

server.TestServerWithTimeseriesImport [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2962246&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2962246&tab=artifacts#/) on master @ [d302ba88430dd2438a5edcee15edf18286f81485](https://github.com/cockroachdb/cockroach/commits/d302ba88430dd2438a5edcee15edf18286f81485):


```
I210508 07:06:31.376390 335683 3@vendor/github.com/cockroachdb/pebble/open.go:305  [n?,pebble] 1068  [JOB 1] WAL created 000002
I210508 07:06:31.376566 336820 3@vendor/github.com/cockroachdb/pebble/table_stats.go:118  [n?,pebble] 1069  [JOB 2] all initial table stats loaded
I210508 07:06:31.387128 335683 1@server/server.go:915  [n?] 1070  monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I210508 07:06:31.390949 335683 server/init.go:196  [n?] 1071  no stores initialized
I210508 07:06:31.390984 335683 server/init.go:197  [n?] 1072  awaiting ``cockroach init`` or join with an already initialized node
I210508 07:06:31.391004 335683 server/init.go:261  [n?] 1073  cluster 8eafa08d-2cd3-4619-858b-b7a80cd4abbd has been created
I210508 07:06:31.391032 335683 server/init.go:262  [n?] 1074  allocated node ID: n1 (for self)
I210508 07:06:31.391054 335683 server/init.go:263  [n?] 1075  active cluster version: 21.1-6
I210508 07:06:31.391127 336867 1@server/server.go:1554  [n1] 1076  connecting to gossip network to verify cluster ID "8eafa08d-2cd3-4619-858b-b7a80cd4abbd"
I210508 07:06:31.391122 335683 gossip/gossip.go:402  [n1] 1077  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35747" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:21 minor_val:1 patch:0 internal:6 > build_tag:"v21.2.0-alpha.00000000-334-gd302ba8843" started_at:1620457591391117953 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:34885" >
I210508 07:06:31.391416 336475 kv/kvserver/closedts/provider/provider.go:135  [ct-closer] 1078  disabling legacy closed-timestamp mechanism; the new one is enabled
I210508 07:06:31.393381 335683 server/node.go:388  [n1] 1079  initialized store s1
I210508 07:06:31.393430 335683 kv/kvserver/stores.go:250  [n1] 1080  read 0 node addresses from persistent storage
I210508 07:06:31.393448 336867 1@server/server.go:1557  [n1] 1081  node connected via gossip
I210508 07:06:31.393480 335683 server/node.go:465  [n1] 1082  started with engine type 2
I210508 07:06:31.393500 335683 server/node.go:467  [n1] 1083  started with attributes []
I210508 07:06:31.393542 335683 1@server/server.go:1684  [n1] 1084  starting https server at 127.0.0.1:39993 (use: 127.0.0.1:39993)
I210508 07:06:31.393576 335683 1@server/server.go:1689  [n1] 1085  starting postgres server at 127.0.0.1:34885 (use: 127.0.0.1:34885)
I210508 07:06:31.393608 335683 1@server/server.go:1691  [n1] 1086  starting grpc server at 127.0.0.1:35747
I210508 07:06:31.393639 335683 1@server/server.go:1692  [n1] 1088  advertising CockroachDB node at 127.0.0.1:35747
W210508 07:06:31.393591 336949 kv/kvserver/store.go:1692  [n1,s1,r6/1:/Table/{SystemCon…-11}] 1087  could not gossip system config: [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r6: replica (n1,s1):1 not lease holder; lease holder unknown
W210508 07:06:31.393591 336949 kv/kvserver/store.go:1692  [n1,s1,r6/1:/Table/{SystemCon…-11}] 1087 +(1) [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r6: replica (n1,s1):1 not lease holder; lease holder unknown
W210508 07:06:31.393591 336949 kv/kvserver/store.go:1692  [n1,s1,r6/1:/Table/{SystemCon…-11}] 1087 +Error types: (1) *roachpb.NotLeaseHolderError
I210508 07:06:31.396605 335683 1@util/log/event_log.go:32  [n1] 1089 ={"Timestamp":1620457591396603417,"EventType":"node_join","NodeID":1,"StartedAt":1620457591391117953,"LastUp":1620457591391117953}
I210508 07:06:31.396690 335683 sql/sqlliveness/slinstance/slinstance.go:252  [n1] 1090  starting SQL liveness instance
I210508 07:06:31.396696 337016 sql/temporary_schema.go:492  [n1] 1091  running temporary object cleanup background job
I210508 07:06:31.397649 336880 kv/kvserver/replica_rangefeed.go:619  [n1,rangefeed=lease,s1,r6/1:/Table/{SystemCon…-11}] 1092  RangeFeed closed timestamp is empty
I210508 07:06:31.397997 337016 sql/temporary_schema.go:527  [n1] 1093  found 0 temporary schemas
I210508 07:06:31.398040 337016 sql/temporary_schema.go:530  [n1] 1094  early exiting temporary schema cleaner as no temporary schemas were found
I210508 07:06:31.398063 337016 sql/temporary_schema.go:531  [n1] 1095  completed temporary object cleanup job
I210508 07:06:31.398087 337016 sql/temporary_schema.go:610  [n1] 1096  temporary object cleaner next scheduled to run at 2021-05-08 07:36:31.396675732 +0000 UTC
I210508 07:06:31.399214 337027 sql/sqlliveness/slstorage/slstorage.go:352  [n1] 1097  inserted sqlliveness session 3161271ee8dd4dca975ebc67df453a42
I210508 07:06:31.399257 337027 sql/sqlliveness/slinstance/slinstance.go:144  [n1] 1098  created new SQL liveness session 3161271ee8dd4dca975ebc67df453a42
I210508 07:06:31.401796 337014 1@util/log/event_log.go:32  [n1] 1099 ={"Timestamp":1620457591396603417,"EventType":"node_join","NodeID":1,"StartedAt":1620457591391117953,"LastUp":1620457591391117953}
I210508 07:06:31.402501 336793 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 1100 ={"Timestamp":1620457591397845917,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I210508 07:06:31.405941 336751 migration/migrationmanager/manager.go:102  [n1,intExec=set-setting,migration-mgr] 1101  no need to migrate, cluster already at newest version
I210508 07:06:31.408279 336751 util/log/event_log.go:32  [n1,intExec=set-setting] 1102 ={"Timestamp":1620457591404677525,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.1-6'"],"SettingName":"version","Value":"21.1-6"}
I210508 07:06:31.418238 337393 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 1103 ={"Timestamp":1620457591415302633,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"1fcdc87b-f66b-4de9-b8f9-bf85b2d1b2d8"}
I210508 07:06:31.421333 337053 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1104 ={"Timestamp":1620457591419452017,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210508 07:06:31.423526 337652 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1105 ={"Timestamp":1620457591421521638,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210508 07:06:31.436923 335683 server/server_sql.go:867  [n1] 1106  done ensuring all necessary startup migrations have run
I210508 07:06:31.437024 337833 jobs/job_scheduler.go:360  [n1] 1107  waiting 2m0s before scheduled jobs daemon start
I210508 07:06:31.438484 336904 3@vendor/github.com/cockroachdb/pebble/db.go:1464  [n1,pebble] 1108  [JOB 3] WAL created 000004
I210508 07:06:31.438716 335683 server/server.go:1907  [n1] 1109  imported 100 ts pairs
I210508 07:06:31.439058 337835 server/auto_upgrade.go:55  [n1] 1110  no need to upgrade, cluster already at the newest version
W210508 07:06:31.439197 336733 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 1111  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
I210508 07:06:31.439461 335683 server/server.go:1907  [n1] 1112  imported 72 ts pairs
    testcluster.go:296: unable to guess the assignment of remaining stores map[1:{}] to nodes map[], needs manual mapping
W210508 07:06:31.441604 336732 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 1113  range lookup of key /Table/20 found only non-matching ranges []; retrying
--- FAIL: TestServerWithTimeseriesImport (0.29s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestServerWithTimeseriesImport PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/server @tbg
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerWithTimeseriesImport.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestServerWithTimeseriesImport [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2967576&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2967576&tab=artifacts#/) on master @ [7fb0684f54abd920e7e392c87bd4ec049c494ed2](https://github.com/cockroachdb/cockroach/commits/7fb0684f54abd920e7e392c87bd4ec049c494ed2):


```
W210511 06:56:37.983040 335369 server/status/runtime.go:322  [n?] 1144  could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I210511 06:56:37.984476 335369 3@vendor/github.com/cockroachdb/pebble/version_set.go:156  [n?,pebble] 1145  [JOB 1] MANIFEST created 000001
I210511 06:56:37.984583 335369 3@vendor/github.com/cockroachdb/pebble/open.go:305  [n?,pebble] 1146  [JOB 1] WAL created 000002
I210511 06:56:37.984790 336578 3@vendor/github.com/cockroachdb/pebble/table_stats.go:118  [n?,pebble] 1147  [JOB 2] all initial table stats loaded
I210511 06:56:38.012923 335369 1@server/server.go:915  [n?] 1148  monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I210511 06:56:38.016826 335369 server/init.go:196  [n?] 1149  no stores initialized
I210511 06:56:38.016875 335369 server/init.go:197  [n?] 1150  awaiting ``cockroach init`` or join with an already initialized node
I210511 06:56:38.016892 335369 server/init.go:261  [n?] 1151  cluster 6e32d9e2-c3b2-42aa-8c1a-a55f8e9b66e5 has been created
I210511 06:56:38.016921 335369 server/init.go:262  [n?] 1152  allocated node ID: n1 (for self)
I210511 06:56:38.016941 335369 server/init.go:263  [n?] 1153  active cluster version: 21.1-8
I210511 06:56:38.017005 335369 gossip/gossip.go:402  [n1] 1155  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:44275" > attrs:<> locality:<tiers:<key:"region" value:"test" > tiers:<key:"dc" value:"dc1" > > ServerVersion:<major_val:21 minor_val:1 patch:0 internal:8 > build_tag:"v21.2.0-alpha.00000000-375-g7fb0684f54" started_at:1620716198017000845 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:36571" >
I210511 06:56:38.017032 336635 1@server/server.go:1554  [n1] 1154  connecting to gossip network to verify cluster ID "6e32d9e2-c3b2-42aa-8c1a-a55f8e9b66e5"
I210511 06:56:38.017595 336639 kv/kvserver/closedts/provider/provider.go:135  [ct-closer] 1156  disabling legacy closed-timestamp mechanism; the new one is enabled
I210511 06:56:38.027043 335369 server/node.go:388  [n1] 1157  initialized store s1
I210511 06:56:38.027113 335369 kv/kvserver/stores.go:250  [n1] 1158  read 0 node addresses from persistent storage
I210511 06:56:38.027169 335369 server/node.go:465  [n1] 1159  started with engine type 2
I210511 06:56:38.027198 335369 server/node.go:467  [n1] 1160  started with attributes []
I210511 06:56:38.027251 335369 1@server/server.go:1684  [n1] 1161  starting https server at 127.0.0.1:36273 (use: 127.0.0.1:36273)
I210511 06:56:38.027291 335369 1@server/server.go:1689  [n1] 1162  starting postgres server at 127.0.0.1:36571 (use: 127.0.0.1:36571)
I210511 06:56:38.027328 335369 1@server/server.go:1691  [n1] 1164  starting grpc server at 127.0.0.1:44275
W210511 06:56:38.027283 336713 kv/kvserver/store.go:1692  [n1,s1,r6/1:/Table/{SystemCon…-11}] 1163  could not gossip system config: [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r6: replica (n1,s1):1 not lease holder; lease holder unknown
W210511 06:56:38.027283 336713 kv/kvserver/store.go:1692  [n1,s1,r6/1:/Table/{SystemCon…-11}] 1163 +(1) [NotLeaseHolderError] lease acquisition attempt lost to another lease, which has expired in the meantime; r6: replica (n1,s1):1 not lease holder; lease holder unknown
W210511 06:56:38.027283 336713 kv/kvserver/store.go:1692  [n1,s1,r6/1:/Table/{SystemCon…-11}] 1163 +Error types: (1) *roachpb.NotLeaseHolderError
I210511 06:56:38.027356 335369 1@server/server.go:1692  [n1] 1165  advertising CockroachDB node at 127.0.0.1:44275
I210511 06:56:38.027595 336635 1@server/server.go:1557  [n1] 1166  node connected via gossip
I210511 06:56:38.031100 335369 1@util/log/event_log.go:32  [n1] 1167 ={"Timestamp":1620716198031098350,"EventType":"node_join","NodeID":1,"StartedAt":1620716198017000845,"LastUp":1620716198017000845}
I210511 06:56:38.031201 335369 sql/sqlliveness/slinstance/slinstance.go:252  [n1] 1168  starting SQL liveness instance
I210511 06:56:38.031681 336591 sql/temporary_schema.go:492  [n1] 1169  running temporary object cleanup background job
I210511 06:56:38.032339 336728 kv/kvserver/replica_rangefeed.go:619  [n1,rangefeed=lease,s1,r6/1:/Table/{SystemCon…-11}] 1170  RangeFeed closed timestamp is empty
I210511 06:56:38.033160 336591 sql/temporary_schema.go:527  [n1] 1171  found 0 temporary schemas
I210511 06:56:38.033206 336591 sql/temporary_schema.go:530  [n1] 1172  early exiting temporary schema cleaner as no temporary schemas were found
I210511 06:56:38.033234 336591 sql/temporary_schema.go:531  [n1] 1173  completed temporary object cleanup job
I210511 06:56:38.033258 336591 sql/temporary_schema.go:610  [n1] 1174  temporary object cleaner next scheduled to run at 2021-05-11 07:26:38.031665533 +0000 UTC
I210511 06:56:38.034738 336762 sql/sqlliveness/slstorage/slstorage.go:352  [n1] 1175  inserted sqlliveness session d5eafd8c05614ca6885d98d01b0cd054
I210511 06:56:38.034784 336762 sql/sqlliveness/slinstance/slinstance.go:144  [n1] 1176  created new SQL liveness session d5eafd8c05614ca6885d98d01b0cd054
I210511 06:56:38.038531 336589 1@util/log/event_log.go:32  [n1] 1177 ={"Timestamp":1620716198031098350,"EventType":"node_join","NodeID":1,"StartedAt":1620716198017000845,"LastUp":1620716198017000845}
I210511 06:56:38.039259 336744 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 1178 ={"Timestamp":1620716198032955372,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I210511 06:56:38.042763 336784 migration/migrationmanager/manager.go:102  [n1,intExec=set-setting,migration-mgr] 1179  no need to migrate, cluster already at newest version
I210511 06:56:38.045454 336784 util/log/event_log.go:32  [n1,intExec=set-setting] 1180 ={"Timestamp":1620716198041438479,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.1-8'"],"SettingName":"version","Value":"21.1-8"}
I210511 06:56:38.055585 337064 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 1181 ={"Timestamp":1620716198052551253,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"9a5cdebd-2669-4310-ab29-f397cc28b688"}
I210511 06:56:38.059158 337377 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1182 ={"Timestamp":1620716198056934510,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210511 06:56:38.061606 337324 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1183 ={"Timestamp":1620716198059384815,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
W210511 06:56:38.069275 336513 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 1184  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W210511 06:56:38.071120 336512 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 1185  range lookup of key /Table/20 found only non-matching ranges []; retrying
I210511 06:56:38.074388 335369 server/server_sql.go:867  [n1] 1186  done ensuring all necessary startup migrations have run
I210511 06:56:38.074521 337587 jobs/job_scheduler.go:360  [n1] 1187  waiting 2m0s before scheduled jobs daemon start
I210511 06:56:38.075271 335369 server/server.go:1907  [n1] 1188  imported 21 ts pairs
    testcluster.go:296: unable to guess the assignment of remaining stores map[1:{}] to nodes map[], needs manual mapping
I210511 06:56:38.079087 337589 server/auto_upgrade.go:55  [n1] 1189  no need to upgrade, cluster already at the newest version
--- FAIL: TestServerWithTimeseriesImport (0.31s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestServerWithTimeseriesImport PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/server @tbg
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestServerWithTimeseriesImport.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -1654,11 +1654,6 @@ func (s *Server) PreStart(ctx context.Context) error {
 		return err
 	}
 
-	// Begin recording time series data collected by the status monitor.
-	s.tsDB.PollSource(
-		s.cfg.AmbientCtx, s.recorder, base.DefaultMetricsSampleInterval, ts.Resolution10s, s.stopper,
-	)
-
 	var graphiteOnce sync.Once
 	graphiteEndpoint.SetOnChange(&s.st.SV, func() {
 		if graphiteEndpoint.Get(&s.st.SV) != "" {
@@ -1867,6 +1862,13 @@ func (s *Server) PreStart(ctx context.Context) error {
 
 	log.Event(ctx, "server initialized")
 
+	// Begin recording time series data collected by the status monitor.
+	// This will perform the first write synchronously, which is now
+	// acceptable.
+	s.tsDB.PollSource(
+		s.cfg.AmbientCtx, s.recorder, base.DefaultMetricsSampleInterval, ts.Resolution10s, s.stopper,
+	)
+
 	return maybeImportTS(ctx, s)
 }
 
@@ -1897,6 +1899,9 @@ func maybeImportTS(ctx context.Context, s *Server) error {
 	b := &kv.Batch{}
 	var n int
 	maybeFlush := func(force bool) error {
+		if n == 0 {
+			return nil
+		}
 		if n < 100 && !force {
 			return nil
 		}

--- a/pkg/server/server_import_ts_test.go
+++ b/pkg/server/server_import_ts_test.go
@@ -19,13 +19,12 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
-	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/ts"
 	"github.com/cockroachdb/cockroach/pkg/ts/tspb"
 	"github.com/cockroachdb/cockroach/pkg/util/humanizeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
-	"github.com/cockroachdb/errors"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
 )
@@ -37,6 +36,7 @@ import (
 // amount of data in there.
 func TestServerWithTimeseriesImport(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 	ctx := context.Background()
 
 	path := filepath.Join(t.TempDir(), "dump.raw")
@@ -64,9 +64,9 @@ func TestServerWithTimeseriesImport(t *testing.T) {
 	defer srv.Stopper().Stop(ctx)
 	cc, err := srv.Servers[0].RPCContext().GRPCUnvalidatedDial(srv.Servers[0].RPCAddr()).Connect(ctx)
 	require.NoError(t, err)
-	// This would time out if we didn't supply a dump. Just the fact that
-	// it returns proves that we ingested at least some time series (or
-	// that we failed to disable time series).
+	// This would fail if we didn't supply a dump. Just the fact that it returns
+	// successfully proves that we ingested at least some time series (or that we
+	// failed to disable time series).
 	bytesDumpedAgain := dumpTSNonempty(t, cc, filepath.Join(t.TempDir(), "dump2.raw"))
 	// We get the same number of bytes back, which serves as proximate proof
 	// that we ingested the dump properly.
@@ -74,24 +74,15 @@ func TestServerWithTimeseriesImport(t *testing.T) {
 }
 
 func dumpTSNonempty(t *testing.T, cc *grpc.ClientConn, dest string) (bytes int64) {
-	// NB: at the time of writing, this never has to retry since we're
-	// persisting first batch of timeseries quickly, but better safe than
-	// sorry.
-	testutils.SucceedsSoon(t, func() error {
-		c, err := tspb.NewTimeSeriesClient(cc).DumpRaw(context.Background(), &tspb.DumpRequest{})
-		require.NoError(t, err)
+	c, err := tspb.NewTimeSeriesClient(cc).DumpRaw(context.Background(), &tspb.DumpRequest{})
+	require.NoError(t, err)
 
-		f, err := os.Create(dest)
-		require.NoError(t, err)
-		require.NoError(t, ts.DumpRawTo(c, f))
-		require.NoError(t, f.Close())
-		info, err := os.Stat(dest)
-		require.NoError(t, err)
-		if info.Size() == 0 {
-			return errors.New("no data dumped")
-		}
-		bytes = info.Size()
-		return nil
-	})
-	return bytes
+	f, err := os.Create(dest)
+	require.NoError(t, err)
+	require.NoError(t, ts.DumpRawTo(c, f))
+	require.NoError(t, f.Close())
+	info, err := os.Stat(dest)
+	require.NoError(t, err)
+	require.NotZero(t, info.Size())
+	return info.Size()
 }

--- a/pkg/ts/db.go
+++ b/pkg/ts/db.go
@@ -157,9 +157,9 @@ func (db *DB) PollSource(
 // start begins the goroutine for this poller, which will periodically request
 // time series data from the DataSource and store it.
 func (p *poller) start() {
+	// Poll once immediately and synchronously.
+	p.poll()
 	_ = p.stopper.RunAsyncTask(context.TODO(), "ts-poller", func(context.Context) {
-		// Poll once immediately.
-		p.poll()
 		ticker := time.NewTicker(p.frequency)
 		defer ticker.Stop()
 		for {

